### ISSUE TITLE
Installed package ggplot2 cannot be found when using the Polyglot API

### ISSUE THREAD

**Issue by edopalmieri** [oracle/fastr#22]
Hi, 

I am tryin gto execute a R script which requires the ggplot2 library. The script starts with : 
`library(ggplot2)`

I have installed the ggplot2 as follows: 

$ Rscript -e "install.packages(\"ggplot2\")"
Content type 'application/octet-stream' length 2213308 bytes (2.1 MB)
* installing *source* package ‘ggplot2’ ...
** package ‘ggplot2’ successfully unpacked and MD5 sums checked
** R
** data
*** moving datasets to lazyload DB
** inst
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (ggplot2)

The downloaded source packages are in
	'/private/var/folders/d5/m03q171j0hbfmpwhtyd_85l400010c/T/Rtmp2ldDvy/downloaded_packages'
Updating HTML index of packages in '.Library'
Making 'packages.html' ... done

I am able to execute the [r-spring sample code](https://medium.com/graalvm/enhance-your-java-spring-application-with-r-data-science-b669a8c28bea), from terminal, which uses ggplot2. 

However when trying to execute my script using a Polyglot Context, I get the following error:

`Error in library(ggplot2) : there is no package called ‘ggplot2’
Failed to execute script: Error in library(ggplot2) : there is no package called ‘ggplot2’`

Do I need to setup/install packages in a different way if I intend to use them from Polyglot Context?
Thanks for your help
Edoardo.

**Comment by steve-s** [comment]
Hi Edoardo, 

I've tried the following with RC5 and RC6:

```
context.eval("R", "library(ggplot2); svg(); print(ggplot(mpg, aes(displ, hwy, colour = class)) + geom_point()); print(svg.string());");
```

and it seems to be working fine. R packages are by default installed into `GRAALVM_HOME/jre/languages/R/library` you can check that there is a subdirectory called `ggplot2` there. In R itself you can use function `.libPaths()` to check in which location is the current R session going to look for R packages, try also running `context.eval("R", "print(.libPaths())");` as a sanity check.

**Comment by edopalmieri** [comment]
Hi Steve,

Thanks for your reply. Following your instructions on locating the installation of the ggplot2 library I noticed that it was being installed in the rc4 folder, whereas I am currently using rc5. This was due to a misconfiguration on my side. Now that I have fixed that however, I get the following error when trying to install ggplot2:

`/../openjdk64-1.8.0.172/bin/Rscript: line 33: /Library/Java/JavaVirtualMachines/graalvm-ce-1.0.0-rc5/Contents/Home/jre/languages/R/bin/RMain: No such file or directory
/../openjdk64-1.8.0.172/bin/Rscript: line 33: exec: /Library/Java/JavaVirtualMachines/graalvm-ce-1.0.0-rc5/Contents/Home/jre/languages/R/bin/RMain: cannot execute: No such file or directory`

I am running RC5 CE, could it be that my installation was not complete/correct?

Thanks
Edoardo.

**Comment by steve-s** [comment]
Hi Edoardo, 

when you run `install.packgaes(...)` in R, it actually starts a new separate R process that does the installation. This looks like FastR is unable to start itself. Could be some versions confusion. Are us running the `install.packages('ggplot2')` in R console, via RScript, or via `context.eval("R", "install.packages(...)")`?

**Comment by edopalmieri** [comment]
Hi Steve,
Thanks for the feedback. I was executing form terminal as instructed here: https://medium.com/graalvm/enhance-your-java-spring-application-with-r-data-science-b669a8c28bea

Having said that, I have since installed RC6 and the problem went away. It must have been an installation problem (both RC5 and RC6 I tried are CE editions). 

ggplot2 and all its dependencies were installed (took quite a while) and I can now run my script, which includes a call to the ward function. In the console output I get the following message:
The "ward" method has been renamed to "ward.D"; note new "ward.D2"

Modifying my script to use ward.D works and the message does not show. Is that an R requirement/change or a Graal "deviation" from the standard language?

Cheers
Edoardo.
